MapboxCoreNavigation/NavigationService.swift

```diff
@@ -342,7 +342,7 @@ public class MapboxNavigationService: NSObject, NavigationService, DefaultInterf
         nativeLocationSource.stopUpdatingHeading()
         nativeLocationSource.stopUpdatingLocation()
         
-        if simulationMode == .always {
+        if [.always, .onPoorGPS].contains(simulationMode) {
```

Comment:
Should we only stop on `onPoorGPS` if we’re actually simulating? Or is it safe to call `endSimulation()` regardless?